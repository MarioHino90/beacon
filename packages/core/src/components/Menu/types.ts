import {
  FlattenInterpolation,
  FlattenSimpleInterpolation,
  ThemedStyledProps,
} from "styled-components";

import { PopperProps } from "../Popper";

export type KeyboardKey =
  | "arrowDown"
  | "arrowUp"
  | "enter"
  | "space"
  | "home"
  | "end"
  | "esc"
  | "letter"
  | "tab";

export type Alphabet =
  | "a"
  | "b"
  | "c"
  | "d"
  | "e"
  | "f"
  | "g"
  | "h"
  | "i"
  | "j"
  | "k"
  | "l"
  | "m"
  | "n"
  | "o"
  | "p"
  | "q"
  | "r"
  | "s"
  | "t"
  | "u"
  | "v"
  | "w"
  | "x"
  | "y"
  | "z";

export type OpenFocusTarget = "first" | "last" | null;

export type ActiveDescendantId = string | undefined;

export interface MenuProps<E = HTMLElement>
  extends Pick<
      StyledMenuProps,
      | "className"
      | "id"
      | "size"
      | "style"
      | "styledCss"
      | "hideTopMargin"
      | "variant"
    >,
    Omit<PopperProps<E>, "tipVariant"> {
  /** indicates that the menu should be closed when a user clicks inside the menu; does NOT prevent event handlers from firing */
  closeOnClickInside?: boolean;
  /** class name to be passed to the parent popper element */
  popperClassName?: string;
  /** DOM id to be passed to the parent popper element */
  popperId?: string;
}

export interface BaseStyledMenuProps
  extends React.HTMLAttributes<HTMLUListElement> {
  /** child elements to be rendered inside the component tree, including the render props option */
  children?:
    | React.ReactNode
    | React.ReactNode[]
    | ((renderProps) => React.ReactNode | React.ReactNode[]);
  /** size of the menu component, primarily affects width */
  size?: "sm" | "lg";
  /** indicates that the menu doesn't have any spacing on top */
  hideTopMargin?: boolean;
  /** different visual treatments depending on menu use case */
  variant?: "default" | "nav";
}

// eslint-disable-next-line @typescript-eslint/ban-types
export interface StyledMenuProps<T extends object = {}>
  extends BaseStyledMenuProps {
  /** prop to interpolate the returned value of a `css` call within styled-components */
  styledCss?:
    | FlattenSimpleInterpolation
    // eslint-disable-next-line @typescript-eslint/ban-types
    | FlattenInterpolation<ThemedStyledProps<{}, T>>
    | FlattenInterpolation<ThemedStyledProps<BaseStyledMenuProps, T>>;
}
