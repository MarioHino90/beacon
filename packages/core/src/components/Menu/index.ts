export { Menu, useMenu } from "./Menu";

export { StyledMenu } from "./StyledMenu";

export type { MenuProps, StyledMenuProps } from "./types";
