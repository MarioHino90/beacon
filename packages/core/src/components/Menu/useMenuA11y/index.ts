export { useMenuA11y } from "./useMenuA11y";
