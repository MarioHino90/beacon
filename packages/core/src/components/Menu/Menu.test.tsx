import React from "react";
import { shallow } from "enzyme";
import BeaconThemeProvider from "@braze/beacon-styling";
import { css } from "styled-components";

import { PlacementType, placementTypes } from "../Popper";

import { MenuProps } from "./types";
import { Menu } from "./Menu";

describe("Menu", () => {
  let props: MenuProps;

  beforeEach(() => {
    props = {
      id: "lime",
      className: "someClassName",
      onOpen: jest.fn(),
      onClose: jest.fn(),
      popperClassName: "someOtherClassName",
      popperId: "someId",
    };
  });

  it.each([...placementTypes, undefined])(
    "should match the snapshot when placement is '%s' and 'isOpen' is false",
    (placement: PlacementType) => {
      const wrapper = shallow(
        <Menu {...props} placement={placement} isOpen={false}>
          <div>Hi</div>
        </Menu>,
        { wrappingComponent: BeaconThemeProvider }
      );

      expect(wrapper).toMatchSnapshot();
    }
  );

  it.each([...placementTypes, undefined])(
    "should match the snapshot when placement is '%s' and 'isOpen' is true",
    (placement: PlacementType) => {
      const wrapper = shallow(
        <Menu {...props} placement={placement} isOpen={true}>
          <div>Hi</div>
        </Menu>,
        { wrappingComponent: BeaconThemeProvider }
      );

      expect(wrapper).toMatchSnapshot();
    }
  );

  it(`should match snapshot when 'styledCss' is defined`, () => {
    const testCss = css`
      background-color: pink;
    `;
    const wrapper = shallow(
      <Menu {...props} styledCss={testCss}>
        <div>Hi</div>
      </Menu>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it(`should match snapshot when 'hideTopMargin' is defined`, () => {
    const wrapper = shallow(
      <Menu {...props} hideTopMargin>
        <div>Hi</div>
      </Menu>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it.each(["sm", "lg"] as MenuProps["size"][])(
    "should match snapshot when 'size' prop is '%s'",
    (size) => {
      const wrapper = shallow(
        <Menu {...props} isOpen={true} size={size}>
          <div>Hi</div>
        </Menu>,
        { wrappingComponent: BeaconThemeProvider }
      );

      expect(wrapper).toMatchSnapshot();
    }
  );
});
