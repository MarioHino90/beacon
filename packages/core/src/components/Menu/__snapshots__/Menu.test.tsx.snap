// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match snapshot when 'hideTopMargin' is defined 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  style={
    {
      "borderTop": 0,
    }
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match snapshot when 'size' prop is 'lg' 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={true}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match snapshot when 'size' prop is 'sm' 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={true}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match snapshot when 'styledCss' is defined 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'auto' and 'isOpen' is false 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={false}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="auto"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'auto' and 'isOpen' is true 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={true}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="auto"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'bottom' and 'isOpen' is false 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={false}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="bottom"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'bottom' and 'isOpen' is true 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={true}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="bottom"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'bottom-end' and 'isOpen' is false 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={false}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="bottom-end"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'bottom-end' and 'isOpen' is true 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={true}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="bottom-end"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'bottom-start' and 'isOpen' is false 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={false}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="bottom-start"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'bottom-start' and 'isOpen' is true 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={true}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="bottom-start"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'left' and 'isOpen' is false 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={false}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="left"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'left' and 'isOpen' is true 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={true}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="left"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'left-end' and 'isOpen' is false 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={false}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="left-end"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'left-end' and 'isOpen' is true 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={true}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="left-end"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'left-start' and 'isOpen' is false 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={false}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="left-start"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'left-start' and 'isOpen' is true 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={true}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="left-start"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'right' and 'isOpen' is false 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={false}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="right"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'right' and 'isOpen' is true 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={true}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="right"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'right-end' and 'isOpen' is false 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={false}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="right-end"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'right-end' and 'isOpen' is true 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={true}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="right-end"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'right-start' and 'isOpen' is false 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={false}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="right-start"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'right-start' and 'isOpen' is true 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={true}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="right-start"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'top' and 'isOpen' is false 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={false}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="top"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'top' and 'isOpen' is true 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={true}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="top"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'top-end' and 'isOpen' is false 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={false}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="top-end"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'top-end' and 'isOpen' is true 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={true}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="top-end"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'top-start' and 'isOpen' is false 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={false}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="top-start"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'top-start' and 'isOpen' is true 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={true}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  placement="top-start"
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'undefined' and 'isOpen' is false 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={false}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;

exports[`Menu should match the snapshot when placement is 'undefined' and 'isOpen' is true 1`] = `
<Popper
  className="someOtherClassName"
  id="someId"
  interactive={true}
  isOpen={true}
  noArrow={true}
  onClose={[MockFunction]}
  onOpen={[MockFunction]}
  popperStyledCss={
    [
      "box-sizing:border-box;max-width:460px;background:transparent;",
    ]
  }
  triggers={
    [
      "click",
    ]
  }
  untriggers={
    [
      "click",
    ]
  }
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <Component />
  </Transition>
</Popper>
`;
